src/System.Text.Json/tests/Utf8JsonWriterTests.cs

```diff
@@ -3630,20 +3630,30 @@ public void EscapeSurrogatePairs(bool formatted, bool skipValidation)
         }
 
         [Fact]
-        public void EscapeSurrogatePairsThrowsHighSurrogateMissing()
+        public void EscapeSurrogatePairs()
         {
             string propertyName = "a \uD800\uDC00\uDE6D a";
             string value = propertyName;
             var output = new ArrayBufferWriter<byte>(12);
-            using var jsonUtf8 = new Utf8JsonWriter(output);
-
-            Assert.Throws<ArgumentException>(() => jsonUtf8.WriteString(propertyName, value));
+            using (var jsonUtf8 = new Utf8JsonWriter(output))
+            {
+                jsonUtf8.WriteStartObject();
+                jsonUtf8.WriteString(propertyName, value);
+                jsonUtf8.WriteEndObject();
+                jsonUtf8.Flush();
+                string result = Encoding.UTF8.GetString(
```

Comment:
Use `JsonTestHelper.AssertContents`?

Also, consider moving this outside the using block (and then we can remove the call to Flush).

Reply:
Sure I'll change it. I'll also add a "todo" for `AssertContents` and `AssertContentsNotEqual` since they assume Json.NET format and thus will do an unnecessary normalization when the expected value is not Json.Net.